sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1059,6 +1060,14 @@ def _replace_query_for_model(self, model: Model, name: str, query_or_df: QueryOr
             column_descriptions=model.column_descriptions,
         )
 
+    def allow_clone_for_dev_preview(self, snapshot: Snapshot) -> bool:
```

Comment:
note: the `snapshot` parameter is currently unused but I left it there incase we wanted evaluation strategies to be able to make more fine-grained decisions in future